File: Semestrul 4/MPP2/Problema8_Csharp_ORM/Persistence/IRepository.cs
using Problema8SC_CSharp.Model;


namespace Problema8SC_CSharp.Persistence;

public interface IRepository<T, ID> where T : Entity<ID>
{
    T FindOne(ID id);
    IEnumerable<T> FindAll();
    void Save(T entity);
    void Delete(ID id);
    void Update(T entity);
}

File: Semestrul 4/MPP2/Problema8_FX/src/main/java/ro/mpp2025/problema8_fx/Repository/IRepository.java
package ro.mpp2025.problema8_fx.Repository;

import ro.mpp2025.problema8_fx.Domain.Entity;

public interface IRepository<T extends Entity<ID>, ID> {
    T findOne(ID id);
    Iterable<T> findAll();
    void save(T entity);
    void delete(ID id);
    void update(T entity);
}
